test/Microsoft.ML.Tests/SelectColumnsTransformsTests.cs

```diff
@@ -0,0 +1,319 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.Transforms;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class SelectColumnsTransformsTests : TestDataPipeBase
+    {
+        class TestClass
+        {
+            public int A;
+            public int B;
+            public int C;
+        }
+
+        public SelectColumnsTransformsTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        void TestSelect()
+        {
+            var data = new[] { new TestClass() { A = 1, B = 2, C = 3, }, new TestClass() { A = 4, B = 5, C = 6 } };
+            using (var env = new ConsoleEnvironment())
+            {
+                var dataView = ComponentCreation.CreateDataView(env, data);
+                var est = new SelectColumnsEstimator(env, new[]{ "A", "C" });
+                var transformer = est.Fit(dataView);
+                var result = transformer.Transform(dataView);
+                var foundColumnA = result.Schema.TryGetColumnIndex("A", out int aIdx);
+                var foundColumnB = result.Schema.TryGetColumnIndex("B", out int bIdx);
+                var foundColumnC = result.Schema.TryGetColumnIndex("C", out int cIdx);
+
+                Assert.True(foundColumnA);
+                Assert.Equal(0, aIdx);
+                Assert.False(foundColumnB);
+                Assert.Equal(0, bIdx);
+                Assert.True(foundColumnC);
+                Assert.Equal(1, cIdx);
+            }
+        }
+        
+        [Fact]
+        void TestSelectWorkout()
+        {
+            var data = new[] { new TestClass() { A = 1, B = 2, C = 3, }, new TestClass() { A = 4, B = 5, C = 6 } };
+            using (var env = new ConsoleEnvironment())
+            {
+                var dataView = ComponentCreation.CreateDataView(env, data);
+                var est = new SelectColumnsEstimator(env, new[]{ "D", "E" });
+                TestEstimatorCore(est, validFitInput: dataView);
```

Comment:
>TestEstimatorCore [](start = 16, length = 17)

actually, let's make it more elaborate, and 
- test the predicate version here (for instance, keep only columns with >1 characters in name)
- add an invalid dataset

Reply:
since the api change, I have TestEstimatorCore for both keep and drop scenarios both with invalid data sets.

---
In reply to: [226068148](https://github.com/dotnet/machinelearning/pull/1269#discussion_r226068148) [](ancestors = 226068148)